### PR TITLE
Don't return or accept `node_version` in the Desired Nodes API

### PR DESCRIPTION
**Pull request by arteam**
It was deprecated in #104209 (8.13) and shouldn't be set or returned in 9.0

The Desired Nodes API is an internal API, and users shouldn't depend on its backward compatibility.

### CODE REVIEW

**Comment by arteam** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java [hunk 1]
```diff
@@ -409,7 +346,7 @@ public String toString() {
     }
 
     public boolean hasVersion() {
-        return Strings.isNullOrBlank(version) == false;
+        return false;
     }
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java:350]: I think it only had 2 usages (have I missed more in dependencies?) that has been removed in this pr.
Is it save to remove this method entirely?

Comment (arteam) [reply server/src/main/java/org/elasticsearch/cluster/metadata/DesiredNode.java:350]: Thank you! Yes, it should be safe to remove the method now :+1:

**Review by mark-vieira — APPROVED** [review]

**Comment by DaveCTurner** [comment]
> The Desired Nodes API is an internal API, and users shouldn't depend on its backward compatibility.

Hi @arteam, this isn't true, we need to preserve the v8 behaviour (i.e. the deprecation warnings) if the user is using the v8 API. Also really this one needs transport version changes so we can't really address it until after v8.last FF. It's going to be tricky to address these issues correctly on top of this change so I'd rather that you reverted it and then we can follow up later on.

**Comment by arteam** [comment]
Waiting until we have TransportVersions_V9 and I will keep the `node_version` field in order to support running the desired nodes API in a V8 compatibility node.

**Pull request merged**